File: Tatsuki_Yabe/Q1/Go/pseudocode_walkthrough.go
package main

func maxArea2(height []int) int {
	l := 0
	r := len(height) - 1
	maxarea := 0

	for l < r {
		length := min(height[l], height[r])
		width := r - l
		area := length * width
		maxarea = max(maxarea, area)

		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}

	return maxarea
}

File: Tatsuki_Yabe/Q1/Swift/pseudocode_walkthrough.swift
class Solution {
    func maxArea2(_ height: [Int]) -> Int {
        var l = 0
        var r = height.count - 1
        var maxarea = 0

        while l < r {
            let length = min(height[l], height[r])
            let width = r - l
            let area = length * width
            maxarea = max(maxarea, area)

            if (l < r) {
                l += 1
            } else {
                r -= 1
            }
        }


        return maxarea
    }
}